targets/esp8266/source/jerry_extapi.c

```diff
@@ -45,46 +44,46 @@ NAME ## _handler (const jerry_api_object_t * function_obj_p __UNSED__, \
 
 DELCARE_HANDLER(assert) {
   if (args_cnt == 1
-      && args_p[0].type == JERRY_API_DATA_TYPE_BOOLEAN
-      && args_p[0].u.v_bool == true)
+      && jerry_value_is_boolean (args_p[0])
+      && jerry_get_boolean_value (args_p[0]))
   {
     printf (">> Jerry assert true\r\n");
-    return true;
+    return jerry_create_boolean (true);
   }
   printf ("Script assertion failed\n");
   exit (JERRY_STANDALONE_EXIT_CODE_FAIL);
-  return false;
+  return jerry_create_boolean (false);
 }
 
 
 DELCARE_HANDLER(print) {
-  jerry_api_length_t cc;
+  jerry_length_t cc;
 
   if (args_cnt)
   {
     printf(">> print(%d) :", (int)args_cnt);
     for (cc=0; cc<args_cnt; cc++)
     {
-      if (args_p[cc].type == JERRY_API_DATA_TYPE_STRING && args_p[cc].u.v_string)
+      if (jerry_value_is_string(args_p[cc]))
       {
         static char buffer[128];
-        jerry_api_size_t length, maxlength;
-        length = -jerry_api_string_to_char_buffer (args_p[0].u.v_string, NULL, 0);
+        jerry_size_t length, maxlength;
+        length = -jerry_string_to_char_buffer (args_p[0], NULL, 0);
```

Comment:
this wont work. This refers to the old behavior. In the past `jerry_string_to_char_buffer` gave back negative number if the buffer was too small, so you where able to allocate a new buffer. Use `jerry_get_string_size` to get the size of string in `args_p[0]`. Rename `length` and `maxlength` to `size` and `maxsize`